### PR TITLE
use `useSWR` data instead calling `setState`

### PR DESCRIPTION
**Pull request by notmd**
I also added some global SWR config.
I didn't refactor the change in #732 cause using custom middleware is more complex and doesn't have any improvement in this case.

### CODE REVIEW

File: website/src/pages/api/auth/[...nextauth].ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ if (boolean(process.env.DEBUG_LOGIN) || process.env.NODE_ENV === "development")
         const user = {
           id: credentials.username,
           name: credentials.username,
+          role: "admin",
```
Comment (notmd) [inline website/src/pages/api/auth/[...nextauth].ts:47]: This is temporal. I will refactor the debug login to allow select roles when login in later PR.

Comment (fozziethebeat) [inline website/src/pages/api/auth/[...nextauth].ts:47]: This seems unrelated.  Is it needed?  the `.env` is configured to set the `admin` debug user with the `admin` role.  We didn't document that anywhere but it should work.

**Review by fozziethebeat — CHANGES_REQUESTED** [review]
Just one small change that I think is handled already by env variables.

**Review by fozziethebeat — APPROVED** [review]
Just saw comment.  Cleanup in a follow up PR is fine.

**Pull request merged**